stablehlo/transforms/StablehloRefineShapes.cpp

```diff
@@ -706,7 +707,18 @@ struct RefineCustomCallOpPattern : public OpRewritePattern<CustomCallOp> {
     SmallVector<ShapedTypeComponents> refinements;
     if (failed(hlo::getShapeRefinements(op.getLoc(), op, refinements)))
       return rewriter.notifyMatchFailure(op, "expected valid refinements");
-    return refineReturnTypes(rewriter, op, refinements);
+    if (failed(refineReturnTypes(rewriter, op, refinements)))
+      return rewriter.notifyMatchFailure(op, "refineReturnTypes failed");
+
+    // Clean up operand buffers after refinement
+    // Must do in this pattern to avoid needing multiple refinement iterations
+    if (op.getCallTargetName().equals(kCustomCallOperandBarrierTarget)) {
+      Value operand = op.getOperand(0);
+      if (operand.getType() == op.getResult(0).getType()) {
+        op.replaceAllUsesWith(ValueRange(operand));
+      }
+    }
+    return success();
```

Comment:
Consider adding some test cases to `stablehlo_refine_shapes.mlir`.

Reply:
Good point. Will do.